services/QuillLMS/client/app/bundles/PremiumHub/containers/UsageSnapshotsContainer.tsx

```diff
@@ -0,0 +1,244 @@
+  const [hasAdjustedFiltersSinceLastSubmission, setHasAdjustedFiltersSinceLastSubmission] = React.useState(null)
+  const [customStartDate, setCustomStartDate] = React.useState(null)
+  const [customEndDate, setCustomEndDate] = React.useState(null)
+  const [searchCount, setSearchCount] = React.useState(0)
+  const [selectedTab, setSelectedTab] = React.useState(ALL)
+  const [showCustomDateModal, setShowCustomDateModal] = React.useState(false)
+  const [lastUsedTimeframe, setLastUsedTimeframe] = React.useState(null)
+  const [showMobileFilterMenu, setShowMobileFilterMenu] = React.useState(false)
+
+  const size = useWindowSize()
+
+  React.useEffect(() => {
+    getFilters()
+  }, [])
+
+  React.useEffect(() => {
+    if (loadingFilters) { return }
+
+    const newValueForHasAdjustedFiltersFromDefault = !unorderedArraysAreEqual(selectedSchools, allSchools) || !unorderedArraysAreEqual(selectedGrades, allGrades) || !unorderedArraysAreEqual(selectedTimeframe, defaultTimeframe(allTimeframes))
+
+    setHasAdjustedFiltersFromDefault(newValueForHasAdjustedFiltersFromDefault)
+
+    const newValueForHasAdjustedFiltersSinceLastSubmission = (!unorderedArraysAreEqual(selectedSchools, lastSubmittedSchools) || !unorderedArraysAreEqual(selectedGrades, lastSubmittedGrades) || !unorderedArraysAreEqual(selectedTimeframe, lastSubmittedTimeframe) || customStartDate !== lastSubmittedCustomStartDate || customEndDate !== lastSubmittedCustomEndDate)
+
+    setHasAdjustedFiltersSinceLastSubmission(newValueForHasAdjustedFiltersSinceLastSubmission)
+  }, [selectedSchools, selectedGrades, selectedTimeframe, customStartDate, customEndDate])
+
+  React.useEffect(() => {
+    if (showCustomDateModal || (customStartDate && customEndDate) || !lastUsedTimeframe) { return }
+
+    setSelectedTimeframe(lastUsedTimeframe)
+  }, [showCustomDateModal])
+
+  function openMobileFilterMenu() { setShowMobileFilterMenu(true) }
+
+  function closeMobileFilterMenu() { setShowMobileFilterMenu(false) }
+
+  function handleSetSelectedTimeframe(timeframe) {
+    setLastUsedTimeframe(selectedTimeframe)
+    setSelectedTimeframe(timeframe)
+
+    if (timeframe.value === CUSTOM) {
+      setShowCustomDateModal(true)
+    }
+  }
+
+  function defaultTimeframe(timeframes) {
+    return timeframes?.find(timeframe => timeframe.default) || null
+  }
+
+  function getFilters() {
+    requestGet('/snapshots/options', (filterData) => {
+      const timeframeOptions = filterData.timeframes.map(tf => ({ ...tf, label: tf.name }))
+      const gradeOptions = filterData.grades.map(grade => ({ ...grade, label: grade.name }))
+      const schoolOptions = filterData.schools.map(school => ({ ...school, label: school.name, value: school.id }))
+
+      const timeframe = defaultTimeframe(timeframeOptions)
+
+      setAllGrades(gradeOptions)
+      setAllTimeframes(timeframeOptions)
+      setAllSchools(schoolOptions)
+      setSelectedGrades(gradeOptions)
```

Comment:
Per our discussion on Slack, I think we want our default filter value for `selectedGrades` to be `[]` in order to avoid filtering out classrooms that don't have an assigned grade.

Reply:
I took a slightly different approach here (`selectedGrades` here remains the same, but when we format it to pass it into the `SnapshotSection` components where the requests actually get made, we pass an empty array if it's full). Otherwise we'd have to do employ much more complicated logic to pass in and out the correct values from the `DropdownInput` component.

Reply:
That seems totally reasonable.